nixpkgs_review/nix.py

```diff
@@ -172,9 +269,79 @@ def nix_build(
     ] + shlex.split(args)
 
     try:
-        sh(command)
-    except subprocess.CalledProcessError:
-        pass
+        proc = sh(command, stderr=subprocess.PIPE)
+        stderr = proc.stderr
+    except subprocess.CalledProcessError as e:
+        stderr = e.stderr
+
+    # Remove boring 'copying path' lines from stderr
+    stderr = "\n".join(
+        line
+        for line in stderr.splitlines()
+        if not (
+            line.startswith("copying path '")
+            and line.endswith("from 'https://cache.nixos.org'...")
+        )
+    )
+
+    has_failed_dependencies = []
+    has_timeout = {}
+    for line in stderr.splitlines():
+        if "dependencies couldn't be built" in line:
+            has_failed_dependencies.append(
+                next(item for item in line.split() if "/nix/store" in item)
```

Comment:
The /nix/store can have some prefix IIRC.

Reply:
Yeah, I'm not sure what the right way to determine what the prefix actually is, if it's not /nix/store. Look at the start of `$(which nix)`? :P

Reply:
You can evaluate `builtins.storeDir`